dashboard/legacy/middleware/helpers/projects.rb

```diff
@@ -136,12 +136,6 @@ def publish(channel_id, type, user)
     project_query_result = @table.where(id: project_id).exclude(state: 'deleted')
     raise NotFound, "channel `#{channel_id}` not found" if project_query_result.empty?
 
-    rails_project = get_rails_project(project_id)
-    if rails_project.apply_project_age_publish_limits?
-      raise PublishError, "User too new to publish channel `#{channel_id}`" unless rails_project.owner_existed_long_enough_to_publish?
```

Comment:
We are now checking these restrictions in `projects_controller.rb` at https://github.com/code-dot-org/code-dot-org/blob/6f9253b46b77e4356b7f7fc678e42dde993b6dec/dashboard/app/controllers/projects_controller.rb#L553-L556 along with other restrictions.

Reply:
@bencodeorg - seeing other tests in `test_channels.rb` that are related to publishing restrictions. What do you think of checking these in `submit` only?

Reply:
Is `submit` the new (only?) path where you can now publish something? Definitely pro moving permissions check into Rails rather than middleware. Even better would be if we could do the permissions check in ability.rb

Reply:
>Is submit the new (only?) path where you can now publish something? 

yes.

>Definitely pro moving permissions check into Rails rather than middleware. Even better would be if we could do the permissions check in ability.rb

Great idea to move to `ability.rb`.

Reply:
It looks like we still have publishing possible via this route, so I think we'd need to keep these checks until this route is removed?

https://github.com/code-dot-org/code-dot-org/blob/e761ce3b4d79474c11a23375548a2fbe6d4b6b67/dashboard/legacy/middleware/channels_api.rb#L207-L220

Reply:
Ahh...good call.

Reply:
https://codedotorg.atlassian.net/browse/LABS-1185